lib/active_shipping/carriers/ups.rb

```diff
@@ -450,15 +450,26 @@ def build_shipment_request(origin, destination, packages, options={})
             end
           end
 
-          # I don't know all of the options that UPS supports for labels
-          # so I'm going with something very simple for now.
+          # Supported image_formats:
+          # GIF, EPL, ZPL, STARPL and SPL
+          label_format = options[:label_format].nil? ? 'GIF' : options[:label_format].upcase
```

Comment:
I just reformatted it. If you don't specify a label format the API will return appropriate error. How does that look?